deliverables/ObligatoriskOppgave2.md

```diff
@@ -0,0 +1,157 @@
+
+## Group dynamics and communication
+
+One primary disadvantage of working as a group, is the difficulty that comes when attempting to distribute the workload is such a manner that everyone has the same amount to do. To combat this disadvantage, we try to make sure that everyone is satisfied or “okay” with the amount of work they are doing relative to the group, we have yet to have a conflict arise from this disadvantage. We are working well together with the chosen methods and are attempting to include everyone to as much of an extent as possible. When there’s a disagreement, everyone presents their case, and we vote on what to do.
+
+Communication via slack has allowed for near seem less planning, discussing, questions/answers, and even the scheduling of extra meetings. Additionally, Slack has made it much easier to get a hold of group leaders or the lecturer if any questions are raised regarding the obligatory assignments or anything else related to the course.
+
+
+## Retrospective and improvements
+
+In retrospect regarding the first two assignments, things are being handled in a timely and professional manner. Even though we are faced with different time schedules in terms of lectures and extra curricular activities, we counter the increased difficulty of scheduling by having an extra meeting every tuesday if necessary, and try to plan have this meeting slot open. Luckily the team is somewhat flexible when it comes to schedules.
+
+
+
+
+Throughout this second assignment, some of the roles were swapped, because Tor Kristian and Tobias were away, they were employed to the secretary role and handled the written prompts for this assignment. This was to make things easier, to combat the fact that it would be hard for them to partake in pair-programming via an online call, as they were with family and unsure of when they could work. Additionally, when Tor Kristian and Tobias had any questions, they asked via Slack.
+
+Prior to the release of the second assignment, we as a group already had made plans of what to do next. We believe that staying ahead a bit will prove to be a smart investment of time and benefit us in future assignments. If we continue development between assignments, the workload when the actual assignment is released is diminished. Even if what we develop isn’t a part of the next assignment, it will still be useful for the final product.
+
+Although the team were flexible during this assignment, Tor Kristian and Tobias will have to make up for their absence with developing in the future, as having them do the written assignments would be unfair to the rest of the team.
+
+
+We should also improve at making sure everyone is working and committing their work to gitHub as it might look like someone is not contributing or contributing less to the project. It's a group project, and everyone should contribute somewhat equally. 
+
```

Comment:
Regarding the list of high level requirements from task 1
* Allthough we wanted to start working on the visual part of the game and get something that is playable, we quickly realized that we need to create a good back-end part of the game first. This is what we ended up working on in this iteration.

Regarding testing and test driven development
* In the first iteration our codebase really only consisted of what we learnt in the guide posted on MittUiB. As the game UI actually needed to run to test anything, we couldnt really write any tests.
In this iteration we have prioritized tests and followed the principles of test driven development. We currently have 24 working tests.